### PR TITLE
deliveries initial logic

### PR DESCRIPTION
**Pull request by Mindaho**

### CODE REVIEW

**Comment by Mindaho** [comment]
Not ready.

File: models/delivery.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+type Status string
+type Type string
+
+type Delivery struct {
+	ID                    uint      `json:"id" example:"1"`
+	CreatedAt             time.Time `json:"createdAt"`
+	RequestedDeliveryTime time.Time `json:"requestedDeliveryTime"`
+	DeliveryCompleteTime  time.Time `json:"deliveryCompleteTime"`
```
Comment (AivarasTaroza) [inline models/delivery.go:24]: Returns "deliveryCompleteTime": "0001-01-01T00:00:00Z" instead of null

File: models/delivery.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	if d.DeliveryType != TypeDelivery && d.DeliveryType != TypeTakeAway {
+		return false
+	}
+	if d.Location == "" {
+		return false
+	}
+	if d.OrderID == 0 {
+		return false
+	}
+
```
Comment (AivarasTaroza) [inline models/delivery.go:48]: Should check delivery status

File: models/delivery.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+type Delivery struct {
+	ID                    uint      `json:"id" example:"1"`
+	CreatedAt             time.Time `json:"createdAt"`
+	RequestedDeliveryTime time.Time `json:"requestedDeliveryTime"`
+	DeliveryCompleteTime  time.Time `json:"deliveryCompleteTime"`
+	Comment               string    `json:"comment"`
+	DeliveryStatus        Status    `json:"deliveryStatus"`
+	DeliveryType          Type      `json:"deliveryType"`
+	Location              string    `json:"location"`
+	OrderID               uint      `json:"-"`
```
Comment (AivarasTaroza) [inline models/delivery.go:29]: When getting the list of all deliveries it would be useful to have the orderId

File: controllers/delivery.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	order.Delivery = &delivery
+
+	if err := order.CalculateTotal(svc.DB); err != nil {
+		return errors.ErrInternalServerError
+	}
+
+	return order
+}
+
+//TODO AddDeliveryReview
```
Comment (AivarasTaroza) [inline controllers/delivery.go:158]: I'll handle the reviews

**Review by AivarasTaroza — CHANGES_REQUESTED** [review]
Also, in CreateOrder, the request with the comment is ignored.

**Comment by AivarasTaroza** [comment]
Oh, and order_dish still doesn't have a copy of profit_margin

File: models/order.go [hunk 5]
```diff
@@ -7,12 +7,11 @@ import (
 )
 
 const (
-	StatusAwaiting   OrderStatus = "awaiting"
-	StatusAccepted   OrderStatus = "accepted"
-	StatusPreparing  OrderStatus = "preparing"
-	StatusDelivering OrderStatus = "delivering"
```
Comment (AivarasTaroza) [inline models/order.go:13]: Not removed in migration

File: models/order.go [hunk 6]
```diff
@@ -7,12 +7,11 @@ import (
...
-	StatusCompleted  OrderStatus = "completed"
-	StatusCanceled   OrderStatus = "canceled"
+	StatusAwaiting  OrderStatus = "awaiting"
+	StatusAccepted  OrderStatus = "accepted"
```
Comment (AivarasTaroza) [inline models/order.go:11]: "accepted" is not in the report.

File: controllers/delivery.go (hunk 4 above)
Comment (AivarasTaroza) [reply controllers/delivery.go:158]: Done on another branch

File: models/order.go (hunk 5 above)
Comment (AivarasTaroza) [reply models/order.go:13]: You should fix this in a new migration file so it would run after the one in the other branch.

**Pull request merged**